api/src/main/java/org/openmrs/module/appointments/service/impl/PatientAppointmentNotifierService.java

```diff
@@ -27,14 +28,27 @@ public PatientAppointmentNotifierService(List<AppointmentEventNotifier> notifier
     }
 
     public List<NotificationResult> notifyAll(final Appointment appointment) {
+        return notifyAll(appointment, null, null, null);
+    }
+
+    public List<NotificationResult> notifyAll(final Patient patient, final String provider, final String link) {
```

Comment:
should take an array of providers. There can be multiple providers for an appointment.

Reply:
@angshu This is for adhoc teleconsultation which a doctor triggers so the provider here is the provider who is initiating the call

Reply:
@angshu : Shall we mark this conversation as resolved?

Reply:
again, my question would be why not take it from the appointment object, why introduce another String to pass "provider" details.